bin/pontoon-to-webext.js

```diff
@@ -0,0 +1,165 @@
+#! /usr/bin/env node
+
+/* eslint-disable promise/avoid-new */
+
+const propertiesParser = require('properties-parser');
+const path = require('path');
+const FS = require('q-io/fs');
+const argv = require('minimist')(process.argv.slice(2));
+
+const Habitat = require('habitat');
+Habitat.load();
+
+const regexPlaceholders = /\{([A-Za-z0-9_@]*)\}/g;
+let supportedLocales = process.env.SUPPORTED_LOCALES || '*';
+
+const config = {
+  'dest': argv.dest || 'dist/_locales',
+  'src': argv.src || 'locales',
+  'default_locale': argv.locale || 'en-US'
+};
+
+function log(...args) {
+  console.log(...args);  // eslint-disable-line no-console
+}
+
+function error(...args) {
+  console.error(...args);  // eslint-disable-line no-console
+}
+
+function getListLocales() {
+  return new Promise((resolve, reject) => {
+    if (supportedLocales === '*') {
+      FS.listDirectoryTree(path.join(process.cwd(), config.src)).then((dirTree) => {
+        const localeList = [];
+
+        // Get rid of the top level, we're only interested with what's inside it
+        dirTree.splice(0,1);
+        dirTree.forEach((localeLocation) => {
+          // Get the locale code from the end of the path. We're expecting the structure of Pontoon's output here
+          const langcode = localeLocation.split(path.sep).slice(-1)[0];
+
+          if (langcode) {
+            localeList.push(langcode);
+          }
+        });
+        return resolve(localeList);
+      }).catch((e) => {
+        reject(e);
+      });
+    } else {
+      supportedLocales = supportedLocales.split(',').map(item => item.trim());
+      resolve(supportedLocales);
+    }
+  });
+}
+
+function writeFiles(entries) {
+  for (const entry of entries) {
+    const publicPath = path.join(process.cwd(), config.dest, entry.locale.replace('-', '_'));
+    const localesPath = path.join(publicPath, 'messages.json');
+
+    FS.makeTree(publicPath).then(() => {
+      return FS.write(localesPath, JSON.stringify(entry.content, null, 2));
+    }).then(() => {
+      log(`Done compiling locales at: ${localesPath}`);
+    }).catch((e) => {
+      error(e);
```

Comment:
These errors won't lead to a build error, but they should.  It's okay if build scripts are a little finicky as long as they fail hard.

I'd recommend:

```js
function fatal(...args) {
  console.error(...args);
  process.exit(1);
}
```

Also this is all kind of brutal in Node.js compared to Python, but eh.